daemon/c_smartcard.c

```diff
@@ -157,12 +159,19 @@ static TokenToDaemon *
 c_smartcard_send_recv_block(const DaemonToToken *out)
 {
 	ASSERT(out);
-
+#ifdef SYSTEMD
+	int sock = sock_sd_listen_fd(SYSTEMD_SCD_CONTROL_SOCKET);
```

Comment:
this seems wrong we do not want the listen socket, we  want to connect as client

Reply:
In case of the cmld.service, Systemd connects to the cml-scd-control socket and passes the corresponding fd to the daemon process. (see 'OpenFile' line in cmld.service)